src/javascript/app_2/Stores/Modules/Trading/Helpers/proposal.js

```diff
@@ -36,11 +36,18 @@ export const createProposalRequests = (store) => {
 
 const createProposalRequestForContract = (store, type_of_contract) => {
     const obj_expiry = {};
+
     if (store.expiry_type === 'endtime') {
+        const server_time = store.root_store.common.server_time;
+        const start_date  = moment.unix(store.start_date || (server_time / 1000)).utc();
         const expiry_date = moment.utc(store.expiry_date);
-        const start_date  = moment.unix(store.start_date || (store.root_store.common.server_time / 1000)).utc();
         const is_same_day = expiry_date.isSame(start_date, 'day');
-        const expiry_time = is_same_day ? store.expiry_time : '23:59:59';
+        let expiry_time;
+
+        if (store.contract_start_type === 'spot' || is_same_day) {
+            expiry_time = is_same_day ? store.expiry_time : moment(server_time).utc().format('HH:mm');
+        }
```

Comment:
it
it would be better to simply remove the if and use 
```
expiry_time = (store.contract_start_type === 'spot' ||  is_same_day) ? store.expiry_time : moment(server_time).utc().format('HH:mm');
 }
```